main.cpp

```diff
@@ -493,19 +604,60 @@ struct DigitalWorkstation
         float sartTimeSelectionPoint = 0.0f;
         float endTimeSelectionPoint = 24.0f;
 
-        void loopSelection(bool loopSelcetion);
-        int tapTempoBpm (int numTap, float time);
-        void moveCursorPos (int beat, int mesure);
+        void loopSelection(bool loopSelection);
+        float tapTempoBpm (int numTap, float time);
+        int moveCursorPos (int beat, int mesure);
     };
 
-    //loop audio and midi sections
-    void loopAudioSection (float loopIn, float loopOut);
-    //record audio and midi performance 
+    float loopAudioSection (float loopIn, float loopOut);
     void recordData (int input, bool isMidi);
-    //mix multiple recordings and tracks into a final produced
     void bounce (int tracks, float startingPoint, float endingPoint);
 };
 
+float DigitalWorkstation::loopAudioSection(float loopIn, float loopOut)
+{
+    return loopIn - loopOut;
+}
+
+void DigitalWorkstation::recordData(int input, bool isMidi)
+{
+    if (isMidi && input > -1)
+    {
+        recStatus = true;
+    }
+    else
+    {
+        recStatus = false;
+    }
```

Comment:
You can simplify this as:
```
recStatus = (isMidi && input > -1);
```
Think about what happens when the stuff in the `()` is true, and how you're basically assigning that state to your `recStatus`
- [ ] revise

Reply:
Make's sense!